### PR TITLE
Add menu and lobby

### PR DESCRIPTION
**Pull request by BartoszChadrys**

### CODE REVIEW

**Review by olekgolus11 — CHANGES_REQUESTED** [review]
This looks reallyyyy gooood

File: Client.py [hunk 1]
```diff
@@ -99,7 +101,36 @@ def handleBombPlantedThisRound(self):
...
+            self.sendPlayerInfo()
+            if menuState == "lobby":
+                self.menu.showLobby()
+                self.drawPlayersInLobby()
+            elif self.menu.showMenu() == "lobby":
+                menuState = "lobby"
+            if self.allPlayersJoined():
+                self.menu.showCountDownTimer()
+                running_menu = False
+            for event in pygame.event.get():
```
[the remark this answers is not in the document]
Comment (BartoszChadrys) [reply Client.py:129]: You are right, I changed it to enum

File: Menu/Button.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+class Button:
+    def __init__(self, pos, textInput, font, baseColor, hoveringColor):
+        self.x = pos[0]
+        self.y = pos[1]
+        self.font = font
+        self.baseColor, self.hoveringColor = baseColor, hoveringColor
+        self.textInput = textInput
```
[the remark this answers is not in the document]
Comment (BartoszChadrys) [reply Menu/Button.py:7]: Yes, I fix it and make it in two seperate lines

File: Menu/Menu.py [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+
+    def drawRulesMainText(self):
+        rulesText = self.getFont(75).render("RULES", True, "white")
+        rulesRect = rulesText.get_rect(center=(CENTER_X_POS, 250))
+
+        self.screen.blit(rulesText, rulesRect)
+
+    def drawBackgroundRectangle(self):
+        height = 500
+        width = 1150
```
[the remark this answers is not in the document]
Comment (BartoszChadrys) [reply Menu/Menu.py:68]: It's hard to depend on window size value, so I just set constants to height and width in constants.py

File: Menu/Menu.py [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+            ruleRect.left = CENTER_X_POS / 6
+            self.screen.blit(ruleText, ruleRect)
+
+    def rules(self):
+        while True:
+            self.drawBackground()
+            self.drawMenuText()
+
+            mousePos = pygame.mouse.get_pos()
+
```
[the remark this answers is not in the document]
Comment (BartoszChadrys) [reply Menu/Menu.py]: True, I fix some of these extra lines

**Review by olekgolus11 — APPROVED** [review]

**Pull request merged**